src/eredis_cluster.erl

```diff
@@ -101,6 +101,50 @@ q(Command) ->
 qk(Command, PoolKey) ->
     query(Command, PoolKey).
 
+%% =============================================================================
+%% @doc Multi node query
+%% @end
+%% =============================================================================
+-spec qmn(redis_pipeline_command()) -> redis_pipeline_result().
+qmn(Commands) ->
+    CommandsByPools = split_by_pools(Commands),
```

Comment:
In my opinion the split_by_pools command should return two parameters:

{CommandsByPools, MappingInfo} = split_by_pools(Commands)

Mapping info would contain the information to reorder the results using indexes rather than using "lists:keyfind". For example:

CommandsByPools: [{pool1, [Command1, Command4, Command6...]},{pool2, [Command2,...]}]
MappingInfo: [{pool1, [1,4,6,8,9]}, {pool2, [2,3,5,7,10]}]

when the results are returned, we could then do a zip with index and result:

[[{1, Result1}, {4, Result4}, {6, Result6}, {8, Result8},{9, Result9}],
[{2, Result2}, {3, Result3}, {5, Result5}, {7, Result7},{10, Result10}]]

We could then reconstruct the result list in O(n), with a custom recursive function that iterates through the N pool results in parallel. (Look at the first element of each pool result, compare index, pick the lowest one, remove it from the list, add it to the result list, etc...)

Or if we are lazy, we could just do lists:flatten, and then lists:keysort.

The problem with the current implementation is that you cannot have 2 times the same command in the pipe (for example if I use the pipeline command to buffer queries to redis, i might have two times the instruction : ["INCR", "ABC"]). The performance will also not be so good if we have lot of keys in the pipe.